toolkit/tools/bldtracker/bldtracker.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Tool to initialize a csv file or record a new timestamp
+// for shell scripts during the image-building process.
+
+package main
+
+import (
+	"os"
+	"time"
+
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/exe"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/logger"
+	"github.com/microsoft/CBL-Mariner/toolkit/tools/internal/timestamp"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	app          = kingpin.New("bldtracker", "A tool that helps track build time of different steps in the makefile.")
+	scriptName   = app.Flag("script-name", "The name of the current tool.").Required().String()
+	stepName     = app.Flag("step-name", "The name of the current step.").Required().String()
+	actionName   = app.Flag("action-name", "The name of the current action.").Default("").String()
+	dirPath      = app.Flag("dir-path", "The folder that stores timestamp csvs.").Required().ExistingDir() // currently must be absolute
+	logPath      = app.Flag("log-path", "Directory for log files").Required().ExistingDir()
+	validModes   = []string{"n", "r"}
+	mode         = app.Flag("mode", "The mode of this tool. Could be 'initialize' ('n') or 'record'('r').").Required().Enum(validModes...)
+	completePath string
+)
+
+func main() {
+	app.Version(exe.ToolkitVersion)
+	kingpin.MustParse(app.Parse(os.Args[1:]))
+	logger.InitBestEffort(*logPath+"/chroot_timestamp.log", "trace")
+
+	// Construct the csv path.
+	completePath = *dirPath + "/" + *scriptName + ".csv"
+
+	// Perform different actions based on the input "mode".
+	switch *mode {
+	case "n":
+		initialize()
+		break
+	case "r":
+		record()
+		break
+	default:
+		logger.Log.Warnf("Invalid call. Mode must be 'n' for initialize or 'r' for record. ")
+	}
+}
+
+// Creates a csv specifically for the shell script mentioned in "scriptName".
+func initialize() {
+	file, err := os.Create(completePath)
+	if err != nil {
+		logger.Log.Warnf("Unable to create file: %s", completePath)
```

Comment:
Since that's the main function of this call, I think we should actually log an error and return from the call. Considering that the tool is fairly short, I'd even suggest using the `logger.Panicf`, which will print the error and terminate the tool in one go.